components/mailgun/sources/common-webhook.js

```diff
@@ -0,0 +1,97 @@
+const mailgun = require("../mailgun.app");
+
+module.exports = {
+  props: {
+    mailgun,
+    webhookSigningKey: {
+      type: "string",
+      secret: true,
+      label: "Mailgun webhook signing key",
+      description:
+        "Your Mailgun webhook signing key, found [in your Mailgun dashboard](https://app.mailgun.com/app/dashboard), located under Settings on the left-hand nav and then in API Keys look for webhook signing key. Required to compute the authentication signature of events."
+    },
+    http: "$.interface.http",
+    db: "$.service.db",
+  },
+  methods: {
+    verify(signingKey, timestamp, token, signature) {
+      const crypto = require("crypto");
+      const encodedToken = crypto
+        .createHmac("sha256", signingKey)
+        .update(timestamp.concat(token))
+        .digest("hex");
+      return encodedToken === signature;
+    },
+    emitEvent(eventPayload) {
+      const eventTypes = this.getEventType();
+      if (eventTypes.includes(eventPayload.event)) {
+        const meta = this.generateMeta(eventPayload);
+        this.$emit(eventPayload, meta);
+      }
+    },
+  },
+  hooks: {
+    async activate() {
+      const webhookNames = this.getEventName();
+      for (let i = 0; i < webhookNames.length; i++) {
+        const webhookName = webhookNames[i];
+        const webhookDetails = await this.mailgun.getWebhookDetails(
+          this.domain,
+          webhookName
+        );
+        if (webhookDetails && webhookDetails.urls) {
```

Comment:
If you use `lodash/get` and run:

```javascript
const urls = get(webhookDetails, 'urls', [])
```

Then you can avoid the `slice` below, and simply push to `urls`:

```javascript
urls.push(this.http.endpoint)
```

Reply:
I looked upon the web examples of using `slice` without arguments as to make the copy of the array and found this statement: 

> No competent programmer would ever make code more complicated than it needs to be. 

Yeah it is much clearer using `lodash/get`. 👍